game-core/src/main/java/games/strategy/triplea/ResourceLocationTracker.java

```diff
@@ -10,42 +15,30 @@
  */
 @UtilityClass
 class ResourceLocationTracker {
+  // All maps must have at least a "baseTiles" folder
+  private static final String REQUIRED_ASSET_EXAMPLE_FOLDER = "baseTiles/";
 
   /**
-   * master zip is the zipped folder format you get when downloading from a map repo via the 'clone
-   * or download' button.
-   */
-  static final String MASTER_ZIP_MAGIC_PREFIX = "-master/map/";
-
-  static final String MASTER_ZIP_IDENTIFYING_SUFFIX = "-master.zip";
-
-  /**
-   * * Will return an empty string unless a special prefix is needed, in which case that prefix is *
-   * constructed based on the map name. * *
-   *
-   * <p>The 'mapPrefix' is the path within a map zip file where we will then find any map contents.
-   * * For example, if the map prefix is "map", then when we expand the map zip, we would expect *
-   * "/map" to be the first folder we see, and we would expect things like "/map/game" and *
-   * "/map/polygons.txt" to exist.
+   * Will return an empty string unless a special prefix is needed, in which case that prefix is *
+   * constructed based on where the {@code baseTiles} folder is located within the zip.
    *
-   * @param mapName Used to construct any special resource loading path prefixes, used as needed
-   *     depending upon which resources are in the path
    * @param resourcePaths The list of paths used for a map as resources. From this we can determine
    *     if the map is being loaded from a zip or a directory, and if zip, if it matches any
    *     particular naming.
    */
-  static String getMapPrefix(final String mapName, final URL[] resourcePaths) {
-    final boolean isUsingMasterZip =
-        Arrays.stream(resourcePaths)
-            .map(Object::toString)
-            .anyMatch(path -> path.endsWith(MASTER_ZIP_IDENTIFYING_SUFFIX));
-
-    // map skins will have the full path name as their map name.
-    if (mapName.endsWith("-master.zip")) {
-      return mapName.substring(0, mapName.length() - "-master.zip".length())
-          + MASTER_ZIP_MAGIC_PREFIX;
-    } else {
-      return isUsingMasterZip ? mapName + MASTER_ZIP_MAGIC_PREFIX : "";
+  static String getMapPrefix(final URL[] resourcePaths) {
+    for (final URL url : resourcePaths) {
+      try (ZipFile zip = new ZipFile(new File(url.toURI()))) {
+        final Optional<? extends ZipEntry> e =
+            zip.stream().filter($ -> $.getName().endsWith(REQUIRED_ASSET_EXAMPLE_FOLDER)).findAny();
+        if (e.isPresent()) {
+          final String path = e.get().getName();
+          return path.substring(0, path.length() - REQUIRED_ASSET_EXAMPLE_FOLDER.length());
+        }
+      } catch (final IOException | URISyntaxException e) {
+        // File is not a zip or can't be opened
```

Comment:
IOException is highly suspicious. We may want to tell a user about this if we hit it. Perhaps we expect this to be an impossible condition, though if the permissions are wrong then it might be worth notifying the user.

If we first check that a file exists: `new File(url.toURI()).exists()`, and then we get an IOException, it seems like that would be a good place to warn a user. WDYT? @jdimeo 

Beyond this last point, this update looks good.

Reply:
@DanVanAtta My mindset here is that this is just auto resource location, not actual loading and parsing. We expect an IOException if the file is a folder (not a zip file) or if there are any other reasons it can't be read, but in that case we should just return a default/assumed path to the resources (like `/map`) not actually throw errors. We can catch and throw errors when we go to actually load and parse the map.

Reply:
Cool, I'm wondering & considering if we are doing control-flow-by-exception-handling. We could check in the `new File(..)` that it is a file, ends with '.zip' and if not then avoid processing the file. This would leave IOException for cases that we do not expect and potentially represent error cases. I'm concerned that by no-op'ing the IOException case that we could be silently suppressing an error that we have not expected.

Reply:
@DanVanAtta What you are describing is exactly what we should do for the actual map loading code. I see this code as very "lazy" - blindly and quickly try to automatically find the resources, purposefully ignoring errors and not being robust. All that you are describing would be appropriate for the actual map parsing code. I wouldn't want a stack trace with a map error to lead a troubleshooter here, since this isn't the "authority" on parsing the map

Reply:
I see, I agree that error handling should be handled upstream. It looks like it is not quite though.. 

From ResourceLoader.java:
```
  private ResourceLoader(final String mapName, final String[] paths) {
    final URL[] urls = new URL[paths.length];
    for (int i = 0; i < paths.length; i++) {
      final File f = new File(paths[i]);
      if (!f.exists()) {
        log.error(f + " does not exist");
      }
      if (!f.isDirectory() && !f.getName().endsWith(".zip")) {
        log.error(f + " is not a directory or a zip file");
      }
      try {
        urls[i] = f.toURI().toURL();
      } catch (final MalformedURLException e) {
        throw new IllegalStateException(e);
      }
    }
    mapPrefix = ResourceLocationTracker.getMapPrefix(mapName, urls);
    // Note: URLClassLoader does not always respect the ordering of the search URLs
    // To solve this we will get all matching paths and then filter by what matched
    // the assets folder.
    loader = new URLClassLoader(urls);
    this.mapName = mapName;
  }
```